dashboard/app/controllers/api_controller.rb

```diff
@@ -259,6 +262,96 @@ def section_assessments
     render json: data
   end
 
+  # Return results for surveys, which are long-assessment LevelGroup levels with the anonymous property.
+  # At least five students in the section must have submitted answers.  The answers for each contained
+  # sublevel are shuffled randomly.
+  def section_surveys
```

Comment:
There's quite a bit going on in this controller method.  Does it make sense to push some of this logic down to the `LevelGroup` model (or a different model)?